sockeye/inference.py

```diff
@@ -303,6 +350,19 @@ def make_input_from_dict(sentence_id: SentenceId,
                 logger.error("There is mismatch in source factors %d and prefix factors %d", len(factors), len(source_prefix_factors))
                 return _bad_input(sentence_id, reason=str(input_dict))
 
+        target_prefix_tokens = input_dict.get(C.JSON_TARGET_PREFIX_KEY)
+        if target_prefix_tokens is not None and not target_prefix_tokens:
```

Comment:
if the target_prefix is a single space (for whatever reason) (`' '`) this will not fire but the target_prefix_tokens will still be empty. How about moving the check into line 357, after the `get_tokens` call:
```
if not target_prefix_tokens:   # none or empty list
   logger.warning(f"Empty string is specified as a target prefix for input '{input_dict[C.JSON_TEXT_KEY]}'.")
```

Reply:
Thanks. I moved the check into line 357 as you suggested and slightly changed it to:

```
target_prefix_tokens = input_dict.get(C.JSON_TARGET_PREFIX_KEY)
target_prefix_tokens = list(utils.get_tokens(target_prefix_tokens)) if target_prefix_tokens is not None else None
if target_prefix_tokens is not None and not target_prefix_tokens:
    logger.warning(f"Empty string is specified as a target prefix for input '{input_dict[C.JSON_TEXT_KEY]}'.")
```
This avoid giving us `warning` when `target_prefix_tokens` is None, which is the case when a ``user" does not specify `JSON_TARGET_PREFIX_KEY` in the JSON input.

Reply:
thanks!